starlette/requests.py

```diff
@@ -216,9 +215,22 @@ async def stream(self) -> typing.AsyncGenerator[bytes, None]:
 
     async def body(self) -> bytes:
         if not hasattr(self, "_body"):
-            chunks = []
-            async for chunk in self.stream():
-                chunks.append(chunk)
+            if hasattr(self, "_headers"):
+                content_length = self.headers.get("Content-Length")
+                if content_length:
+                    # Allocating a bytearray with a fixed size,
+                    # that matches "content-length" header.
```

Comment:
"fixed size" is not technically correct: the bytearray could expand to a larger size.

I'd suggest a comment like: `# Pre-allocate a bytearray large enough to fit "content-length" bytes.`

Reply:
Sounds good.
I'll commit it.